app/src/main/java/com/example/reduxtestapp/domain/model/country/CountryModel.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.reduxtestapp.domain.model.country
+
+import com.example.reduxtestapp.data.model.country.CountryDto
+import com.example.reduxtestapp.data.model.country.CurrencyDto
+import com.example.reduxtestapp.data.model.country.NameDto
+
+data class CountryModel(
+    val name: NameModel = NameModel(),
+    val capitals: List<String> = listOf(),
+    val currencies: Map<String, CurrencyModel> = hashMapOf(),
+    val languages: Map<String, String> = hashMapOf(),
+    val population: Int = 0,
+    val flags: Map<String, String> = hashMapOf(),
+    val translations: Map<String, NameModel> = hashMapOf()
+)
+
+data class NameModel (
+    val common: String = "",
+    val official: String = ""
+)
+
+data class CurrencyModel (
+    val name: String,
+    val symbol: String
+)
+
+fun NameDto.asDomain() = NameModel(
+    common = common,
+    official = official
+)
+
+fun CurrencyDto.asDomain() = CurrencyModel(
+    name = name,
+    symbol = symbol
+)
+
+
+
+fun CountryDto.asDomain() = CountryModel(
+    name = name.asDomain(),
+    capitals = capitals,
+    currencies = currencies.mapValues { it.value.asDomain() },
+    languages = languages,
+    population = population,
+    flags = flags,
+    translations = translations.mapValues { it.value.asDomain() }
+)
+
+fun List<CountryDto>.asDomain() = map { it.asDomain() }
```

Comment:
domain doesn't know about data models so move all data -> domain mappers to `data`